python/cryptol/bv.py

```diff
@@ -0,0 +1,385 @@
+
+from functools import reduce
+from typing import Any, List, Union
+
+class BV:
```

Comment:
A few thoughts:

1) I landed on `BV` because I am of the opinion it should be easy and concise to write these sorts of bit sequence values in programs so they can be clearly incorporated into Cryptol/Python verification scripts and be _preferred_ over using the terser but less precise python native values we happen to support as well (like `bytes` or `int`),

2) Is the abbreviation `BV` actually obscure if you're using Cryptol? I would assume the first guess would be it means "bit vector", but I am assuming of course and am biased by lots of recent exposure to SMTLIB bit vector functions whose names all share the `bv` prefix ,

3) In addition to considering using a longer and possibly clearer name like `BitSequence` or `BitVec`  if we don't like `from cryptol.bv import BV`, we could

    i) give the _module_ a clearer name like `bitvec.py`, so that `from cryptol.bitvec import BV` is perhaps more self explanatory, or

    ii) We could consider using the smallest whole word I can think of in this space for the class name: `Bits`. If I were using this class a lot I think I would prefer just calling it `BV` so code and expressions were easier to read and caused less right-creep... but that's just me =)

Reply:
I think that renaming the module will help quite a bit. What about `cryptol.bitvector`, though?